packages/settingeditor/src/pluginlist.tsx

```diff
@@ -237,10 +250,112 @@ export class PluginList extends ReactWidget {
+   */
+  getFilterString(
+    filter: (item: string) => boolean,
+    props: ISettingRegistry.IProperty,
+    definitions?: any,
+    ref?: string
+  ): string[] {
+    // If properties given are references, populate properties
+    // with corresponding definition.
+    if (ref && definitions) {
+      ref = ref.replace('#/definitions/', '');
+      props = definitions[ref] ?? {};
+    }
+
+    // If given properties are an object, advance into the properties
+    // for that object instead.
+    if (props.properties) {
+      props = props.properties;
+      // If given properties are an array, advance into the properties
+      // for the items instead.
+    } else if (props.items) {
+      props = props.items as any;
+      // Otherwise, you've reached the base case and don't need to check for matching properties
+    } else {
+      return [];
+    }
+
+    // If reference found, recurse
+    if (props['$ref']) {
+      return this.getFilterString(
+        filter,
+        props,
+        definitions,
+        props['$ref'] as string
+      );
+    }
+
+    // Make sure props is non-empty before calling reduce
+    if (Object.keys(props).length === 0) {
+      return [];
+    }
+
+    // Iterate through the properties and check for titles / descriptions that match search.
+    return reduce(
+      Object.keys(props),
+      (acc: string[], value: any) => {
+        // If this is the base case, check for matching title / description
+        const subProps = props[value] as PartialJSONObject;
+        if (
+          !subProps &&
+          (filter((props.title as string)?.toLocaleLowerCase() ?? '') ||
+            filter((props.description as string)?.toLocaleLowerCase() ?? '') ||
+            filter(value))
+        ) {
+          acc.push(props.title ?? value);
+          return acc;
+        }
+
+        // If there are properties in the object, check for title / description
+        if (
+          filter((subProps.title as string)?.toLocaleLowerCase() ?? '') ||
+          filter((subProps.description as string)?.toLocaleLowerCase() ?? '') ||
+          filter(value.toLocaleLowerCase())
+        ) {
+          acc.push(subProps.title ?? value);
+        }
+
+        // Finally, recurse on the properties left.
+        acc.concat(
+          this.getFilterString(
+            filter,
+            subProps as ISettingRegistry.IProperty,
+            definitions,
+            subProps['$ref'] as string
+          )
+        );
+        return acc;
+      },
+      []
+    );
+  }
+
+  /**
+   * Updates the filter when the search bar value changes.
+   * @param filter Filter function passed by search bar based on search value.
+   */
   setFilter(filter: (item: string) => boolean): void {
-    this._filter = (value: ISettingRegistry.IPlugin) => {
-      return filter(value.schema.title?.toLowerCase() ?? '');
+    this._filter = (plugin: ISettingRegistry.IPlugin): string[] | boolean => {
```

Comment:
I would use a signature `(plugin: ISettingRegistry.IPlugin): string[] | null`

So you don't have to test for the return type.